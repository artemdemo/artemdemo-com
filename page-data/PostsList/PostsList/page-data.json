{"componentChunkName":"component---src-pages-posts-list-posts-list-jsx","path":"/PostsList/PostsList/","result":{"pageContext":{}},"staticQueryHashes":[]}