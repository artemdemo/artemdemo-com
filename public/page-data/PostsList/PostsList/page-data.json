{"componentChunkName":"component---src-pages-posts-list-posts-list-jsx","path":"/PostsList/PostsList/","webpackCompilationHash":"8239ad51e369f187ac58","result":{"pageContext":{"isCreatedByStatefulCreatePages":true}}}