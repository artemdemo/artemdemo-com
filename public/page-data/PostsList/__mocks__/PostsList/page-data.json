{"componentChunkName":"component---src-pages-posts-list-mocks-posts-list-jsx","path":"/PostsList/__mocks__/PostsList/","webpackCompilationHash":"75d0da1a0948445d1f69","result":{"pageContext":{"isCreatedByStatefulCreatePages":true}}}