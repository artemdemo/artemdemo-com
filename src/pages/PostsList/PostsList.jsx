import React from 'react';
import _get from 'lodash/get';
import Posts from '../../components/Posts/Posts';
import Pagination from '../../components/Pagination/Pagination';
import BasePage from '../BasePage/BasePage';
import * as utils from '../../services/utils';

class PostsList extends BasePage {
    renderPagination(prefixer = utils.prefixBlog) {
        const totalCount = _get(this.props, 'data.allMarkdownRemark.totalCount', 0);
        const skip = _get(this.props, 'pageContext.skip', 0);
        const limit = _get(this.props, 'pageContext.limit', 0);

        return (
            <Pagination
                skip={skip}
                limit={limit}
                totalPosts={totalCount}
                prefixer={prefixer}
            />
        );
    }

    renderBeforeContent() {
        const posts = _get(this.props, 'data.allMarkdownRemark.edges');
        return (
            <Posts list={posts} />
        );
    }

    renderAfterContent() {
        return this.renderPagination();
    }
}

export default PostsList;
