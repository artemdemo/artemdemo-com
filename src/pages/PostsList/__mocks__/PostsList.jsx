import React from 'react';

class PostsList extends React.PureComponent {
    render() {
        return (
            <div data-mock='PostsList'>
                {this.props.children}
            </div>
        );
    }
}

export default PostsList;