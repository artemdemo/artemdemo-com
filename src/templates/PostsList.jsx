import React from 'react';
import { graphql } from 'gatsby';
import _get from 'lodash/get';
import Bio from '../containers/Bio';
import Layout from '../components/Layout/Layout';
import Posts from '../components/Posts/Posts';
import Pagination from '../components/Pagination/Pagination';

class PostsList extends React.Component {
    render() {
        const siteTitle = _get(this.props, 'data.site.siteMetadata.title');
        const siteDescription = _get(this.props, 'data.site.siteMetadata.description');
        const posts = _get(this.props, 'data.allMarkdownRemark.edges');
        const totalCount = _get(this.props, 'data.allMarkdownRemark.totalCount', 0);

        return (
            <Layout
                title={siteTitle}
                metaTitle={siteTitle}
                description={siteDescription}
                location={this.props.location}
            >
                <Bio />
                <Posts list={posts} />
                <Pagination />
            </Layout>
        )
    }
}

export default PostsList

export const pageQuery = graphql`
  query BlogQuery($skip: Int!, $limit: Int!) {
    site {
      siteMetadata {
        title
        description
      }
    }
    allMarkdownRemark(
      sort: { fields: [frontmatter___date], order: DESC }
      skip: $skip
      limit: $limit
    ) {
      totalCount
      edges {
        node {
          excerpt
          fields {
            slug
          }
          frontmatter {
            date(formatString: "DD MMMM, YYYY")
            title
          }
        }
      }
    }
  }
`;
