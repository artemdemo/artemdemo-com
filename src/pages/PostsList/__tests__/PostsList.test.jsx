import React from 'react';
import renderer from 'react-test-renderer';
import PostsList from '../PostsList';

jest.mock('../../../components/Layout/Layout');
jest.mock('../../../components/Posts/Posts');
jest.mock('../../../components/Pagination/Pagination');

describe('PostsList', () => {
    it('should render', () => {
        const tree = renderer.create(
            <PostsList
                data={{
                    site: {
                        siteMetadata: {
                            title: 'Some title',
                            description: 'Some description',
                        },
                    },
                    allMarkdownRemark: {
                        edges: [
                            {
                                frontmatter: {
                                    title: 'Some title',
                                    date: '10 May, 2018',
                                },
                                fields: {
                                    slug: '/first',
                                },
                                excerpt: 'Excerpt text',
                            }
                        ],
                    },
                }}
                location={{}}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});