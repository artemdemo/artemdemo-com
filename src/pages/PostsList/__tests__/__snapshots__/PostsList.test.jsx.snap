// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsList should render 1`] = `
<div
  data-description="Some description"
  data-metaTitle="Some title"
  data-mock="Layout"
  data-title="Some title"
>
  <div
    data-mock="Posts"
  >
    {
  "list": [
    {
      "frontmatter": {
        "title": "Some title",
        "date": "10 May, 2018"
      },
      "fields": {
        "slug": "/first"
      },
      "excerpt": "Excerpt text"
    }
  ]
}
  </div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": null,
      }
    }
  />
  <div
    data-mock="Pagination"
  >
    {
  "skip": 0,
  "limit": 0,
  "totalPosts": 0
}
  </div>
</div>
`;
